src/Types.hs

```diff
@@ -328,3 +334,19 @@ typesDeleterFunctionType memberType = FuncTy [memberType] UnitTy StaticLifetimeT
 -- | The type of environments sent to Lambdas (used in emitted C code)
 lambdaEnvTy :: Ty
 lambdaEnvTy = StructTy (ConcreteNameTy "LambdaEnv") []
+
+createStructName :: [String] -> String -> String
+createStructName path name = (intercalate "." (path ++ [name]))
+
+getStructName :: Ty -> String
+getStructName (StructTy (ConcreteNameTy name) _) = name
+getStructName (StructTy (VarTy name) _) = name
+getStructName _ = ""
+
+getPathFromStructName :: String -> [String]
```

Comment:
Does this mean that the struct type is stored with it's enclosing environments as part of the name for the type?

(e.g. `(defmodule A (defmodule B (deftype T)))` would create a type with the name "A.B.T")

Reply:
Yep, that's precisely correct.

I think ideally we wouldn't need to do this, but right now there are some functions that take a struct name as an argument and *then* use the name to perform an additional lookup--(see Scoring.hs) in particular. Possibly these can be refactored and eventually the name of the type can drop the module prefixes and we can rely on lookups alone to disambiguate.

Reply:
Seems like `ConcreteNameTy` maybe should contain a `SymPath`..?

Reply:
Yes, that seems reasonable!